packages/zoe/src/contracts/publicAuction.js

```diff
@@ -11,119 +11,139 @@ import {
-
-    // seller will use 'Asset' and 'Ask'. buyer will use 'Asset' and 'Bid'
-    assertKeywords(harden(['Asset', 'Ask']));
-
-    const bidderOfferHook = offerHandle => {
-      // Check that the item is still up for auction
-      if (!zcf.isOfferActive(sellerOfferHandle)) {
-        const rejectMsg = `The item up for auction is not available or the auction has completed`;
-        throw rejectOffer(offerHandle, rejectMsg);
-      }
-      if (allBidHandles.length >= numBidsAllowed) {
-        throw rejectOffer(offerHandle, `No further bids allowed.`);
-      }
-      if (
-        !canTradeWithMapKeywords(sellerOfferHandle, offerHandle, [
-          ['Asset', 'Ask'],
-          ['Asset', 'Bid'],
-        ])
-      ) {
-        const rejectMsg = `Bid was under minimum bid or for the wrong assets`;
-        throw rejectOffer(offerHandle, rejectMsg);
-      }
-
-      // Save valid bid and try to close.
-      allBidHandles.push(offerHandle);
-      if (allBidHandles.length >= numBidsAllowed) {
-        closeAuction(zcf, {
-          auctionLogicFn: secondPriceLogic,
-          sellerOfferHandle,
-          allBidHandles,
-        });
-      }
-      return defaultAcceptanceMsg;
-    };
-
-    const bidderOfferExpected = harden({
-      give: { Bid: null },
-      want: { Asset: null },
-    });
-
-    const makeBidderInvite = () =>
-      zcf.makeInvitation(
-        checkHook(bidderOfferHook, bidderOfferExpected),
-        'bid',
-        harden({
-          customProperties: {
-            auctionedAssets,
-            minimumBid,
-          },
-        }),
-      );
-
-    const sellerOfferHook = offerHandle => {
-      if (auctionedAssets) {
-        throw rejectOffer(offerHandle, `assets already present`);
-      }
-      // Save the valid offer
-      sellerOfferHandle = offerHandle;
-      const { proposal } = zcf.getOffer(offerHandle);
-      auctionedAssets = proposal.give.Asset;
-      minimumBid = proposal.want.Ask;
-      return defaultAcceptanceMsg;
-    };
-
-    const sellerOfferExpected = harden({
-      give: { Asset: null },
-      want: { Ask: null },
-    });
-
-    const makeSellerInvite = () =>
-      zcf.makeInvitation(
-        checkHook(sellerOfferHook, sellerOfferExpected),
-        'sellAssets',
-      );
-
-    zcf.initPublicAPI(
+/**
+ * An auction contract in which the seller offers an Asset for sale, and states
+ * a minimum price. A pre-announced number of bidders compete to offer the best
+ * price. When the appropriate number of bids have been received, the second
+ * price rule is followed, so the highest bidder pays the amount bid by the
+ * second highest bidder.
+ *
+ * makeInstance() specifies the issuers and terms ({ numBidsAllowed }) specify
+ * the number of bids required. An invitation for the seller is returned. The
+ * seller's offer should look like
+ * { give: { Asset: asset }, want: { Ask: minimumBidAmount } }
+ * The asset can be non-fungible, but the Ask amount must be of a fungible
```

Comment:
Is this true? I think it might be able to be nonfungible, even if that doesn't necessary make sense. I think it's technically possible.

Reply:
secondPriceLogic would run and produce an outcome with non-fungible amounts, but the outcome would depend on the order of the bids. I'll change it to 'should' unless you think it's worth explaining.

Reply:
hmm, why do you think the outcome would depend on the order of the bids? isGTE is coherent for nonfungible amounts

Reply:
I think I would leave out mention of fungible/nonfungible for this contract unless it really won't work with nonfungible

Reply:
> hmm, why do you think the outcome would depend on the order of the bids? isGTE is coherent for nonfungible amounts

I thought about this for the case of set amountMath. `isGTE` is defined as sets that contain at least the same items, and possibly others.  so a minimum bid of `(A)` could be met by `(A, B)` or `(A, C)`, but not by `(D)`, no matter how much better `D` is than `A`. If the first bid is `(A, B)`, then `(A, C, D, F)` isn't better  by `isGTE` than `(A, B)`.  Similarly, if the first bid is `(A, C)`, then no one who doesn't have a `C` to offer can offer more (according to `isGTE`) than the initial bid, no matter how high they pile their basket.

isGTE is coherent, but the non-fungible set types I can think of don't allow for the kind of competitive bidding I think we're looking for. Are there non-fungible examples where bidding would work?

Reply:
that makes sense to me. Saying it should be fungible sounds good!